### PR TITLE
Update Project Sample Table to use new metadata fields.

### PR DESCRIPTION
**Pull request by MarkAZhang**
Changed Column Header Dropdown to only show unselected columns.
For now, restricting to certain columns as discussed.

Download by CSV and filter by Sample Type also work.

Renamed Tissue Type to Sample Type for consistency.

Also add explicit options for Sample Type (i.e. Stool, Whole Blood, etc.)

----

Column Dropdowns now look like this:
screen shot 2018-11-26 at 1 22 18 pm

Sample Type Filter Dropdown works with the new metadata and only shows valid options:
screen shot 2018-11-26 at 1 22 07 pm

Sample Table Download also works with the new metadata:
screen shot 2018-11-26 at 1 22 23 pm

### CODE REVIEW

File: app/helpers/samples_helper.rb [hunk 1]
```diff
@@ -206,8 +196,30 @@ def get_total_runtime(pipeline_run, run_stages)
...
+
+    # Use a set to speed up query.
+    query_set = query.to_set
+
+    include_not_set = query.include?('Not set')
+
+    sample_type_metadatum = Metadatum
+                            .where(sample_id: samples.pluck(:id), key: "sample_type")
+
+    if include_not_set
```
[the remark this answers is not in the document]
Comment (MarkAZhang) [reply app/helpers/samples_helper.rb]: My code was confusing. Rewrote it to be more clear.

[the remark this answers is not in the document]
Comment (MarkAZhang) [reply app/helpers/samples_helper.rb]: Let me know if that makes more sense.

File: app/helpers/samples_helper.rb [hunk 2]
```diff
@@ -345,4 +366,15 @@ def get_ready_sample_ids(samples)
 
     ready_sample_ids
   end
+
+  def get_distinct_sample_types(samples)
+    types = Metadatum
+            .where(key: "sample_type")
+            .where(sample_id: samples.pluck(:id))
+            .map(&:validated_value)
```
[the remark this answers is not in the document]
Comment (cdebourcy) [reply app/helpers/samples_helper.rb]: wait never mind

Comment (cdebourcy) [inline app/helpers/samples_helper.rb]: would use `pluck(:text_validated_value)` instead of `map` here for performance

Comment (MarkAZhang) [reply app/helpers/samples_helper.rb]: Fixed, thanks!

**Pull request merged**